pkg/controller/connectionsecret/connectionsecrets.go

```diff
@@ -178,6 +143,34 @@ func RemoveStaleSecretsByUserName(ctx context.Context, k8sClient client.Client,
 	return lastError
 }
 
+func FillPrivateConns(conn deployment.Connection, data *ConnectionData) {
+	if conn.PrivateURL != "" {
+		data.PrivateConnURLs = append(data.PrivateConnURLs, PrivateLinkConnURLs{
+			PvtConnURL:    conn.PrivateURL,
+			PvtSrvConnURL: conn.SrvPrivateURL,
+		})
+	}
+
+	if conn.Serverless {
+		for _, pe := range conn.PrivateEndpoints {
+			data.PrivateConnURLs = append(data.PrivateConnURLs, PrivateLinkConnURLs{
+				PvtSrvConnURL: pe.ServerURL,
+			})
+		}
+	} else {
+		for _, pe := range conn.PrivateEndpoints {
+			data.PrivateConnURLs = append(data.PrivateConnURLs, PrivateLinkConnURLs{
+				PvtConnURL:      pe.URL,
+				PvtSrvConnURL:   pe.ServerURL,
+				PvtShardConnURL: pe.ShardURL,
+			})
+		}
+	}
+}
+
+// FillPrivateConnStrings fills private conn urls from connection strings
+// TODO: remove once all usages move over to FillPrivateConns instead
```

Comment:
is there a ticket for this? if not, please create and attach `TODO(CLOUDP-...)`

Reply:
will double check if this still applies

Reply:
next change refers to CLOUDP-253951